src/components/generic/ResourceTree.tsx

```diff
@@ -99,35 +93,35 @@ class ResourceTree extends React.Component<Props, State> {
    * @param prevProps previous component properties
    */
   public componentDidUpdate = async (prevProps: Props) => {
-    const { resources, selectedContentVersionId, selectedResource } = this.props;
-    const { lockedResourceIds } = this.state;
+    const { resources, selectedResource, lockedResourceIds } = this.props;
 
-    if (prevProps.selectedResource && !selectedResource) {
-      await this.releaseLock(prevProps.selectedResource);
 
-      this.setState({
-        lockedResourceIds: [ ...lockedResourceIds ].filter(id => id !== prevProps.selectedResource!.id)
-      }, () => this.setState({ treeData: this.buildTree(resources) }));
-
-      this.resourceLockInterval && clearInterval(this.resourceLockInterval);
-      return;
-    }
+    const selectedResourceChanged = (!deepEqual(prevProps.selectedResource, selectedResource)) || prevProps.selectedResource?.id !== selectedResource?.id;
```

Comment:
Can the additional parentheses be removed?
```suggestion
    const selectedResourceChanged = !deepEqual(prevProps.selectedResource, selectedResource) || prevProps.selectedResource?.id !== selectedResource?.id;
```